sources/npmRegistryUtils.ts

```diff
@@ -0,0 +1,51 @@
+import {UsageError}          from 'clipanion';
+import {OutgoingHttpHeaders} from 'http2';
+
+import * as httpUtils        from './httpUtils';
+
+// load abbreviated metadata as that's all we need for these calls
+// see: https://github.com/npm/registry/blob/master/docs/responses/package-metadata.md
+export const DEFAULT_HEADERS: OutgoingHttpHeaders = {
+  [`Accept`]: `application/vnd.npm.install-v1+json`,
+};
+export const DEFAULT_NPM_REGISTRY_URL = `https://registry.npmjs.org`;
+
+export async function fetchAsJson(packageName: string) {
+  const npmRegistryUrl = process.env.COREPACK_NPM_REGISTRY || DEFAULT_NPM_REGISTRY_URL;
+
+  if (process.env.COREPACK_ENABLE_NETWORK === `0`)
+    throw new UsageError(`Network access disabled by the environment; can't reach npm repository ${npmRegistryUrl}`);
+
+  const headers = {...DEFAULT_HEADERS};
+
+  if (`COREPACK_NPM_TOKEN` in process.env) {
+    headers.authorization = `Bearer ${process.env.COREPACK_NPM_TOKEN}`;
+  } else if (`COREPACK_NPM_USERNAME` in process.env
+          && `COREPACK_NPM_PASSWORD` in process.env) {
+    const encodedPassword = Buffer.from(`${process.env.COREPACK_NPM_PASSWORD}`, `base64`).toString(`utf8`);
```

Comment:
We expect the password to be provided as base64? I think it should be documented somehow.

Reply:
@aduh95 ah good catch, I didn't even think this through - I adopted the same process that was present in the npm flow. I don't actually see the need for this so I've updated it in my latest push.